development/tools/cmd/secretspopulator/main.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+
+	"cloud.google.com/go/storage"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/cloudkms/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	typedv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const fileNameExtension = "encrypted"
+
+func main() {
+	argBucket := flag.String("bucket", "", "")
+	argKeyring := flag.String("keyring", "", "")
+	argKey := flag.String("key", "", "")
+	argLocation := flag.String("location", "", "")
+	argKubeconfigPath := flag.String("kubeconfig", "", "")
+	argProject := flag.String("project", "", "")
+	argSecretsDefFile := flag.String("secrets-def-file", "", "")
+	flag.Parse()
+	panicOnMissingArg("bucket", argBucket)
+	panicOnMissingArg("keyring", argKeyring)
+	panicOnMissingArg("key", argKey)
+	panicOnMissingArg("location", argLocation)
+	panicOnMissingArg("kubeconfig", argKubeconfigPath)
+	panicOnMissingArg("project", argProject)
+	panicOnMissingArg("secrets-def-file", argSecretsDefFile)
+
+	k8sConfig, err := clientcmd.BuildConfigFromFlags("", *argKubeconfigPath)
+	panicOnError(err)
+	k8sCli, err := kubernetes.NewForConfig(k8sConfig)
+	panicOnError(err)
+
+	secretInterface := k8sCli.CoreV1().Secrets(metav1.NamespaceDefault)
+
+	ctx := context.Background()
+	storageClient, err := storage.NewClient(ctx)
+	panicOnError(err)
+	client, err := google.DefaultClient(ctx, cloudkms.CloudPlatformScope)
+	panicOnError(err)
+
+	kmsService, err := cloudkms.New(client)
+	panicOnError(err)
+
+	p := SecretsPopulator{
+		secretsClient: secretInterface,
+		storageClient: storageClient,
+		kmsClient:     kmsService,
+	}
+
+	secrets, err := readSecretDefFile(*argSecretsDefFile)
+	panicOnError(err)
+
+	panicOnError(p.PopulateSecrets(ctx, *argProject, secrets, *argBucket, *argKeyring, *argKey, *argLocation))
+
+}
+
+// SecretsPopulator is responsible for populating secrets
+type SecretsPopulator struct {
+	secretsClient typedv1.SecretInterface
+	storageClient *storage.Client
+	kmsClient     *cloudkms.Service
+}
+
+// PopulateSecrets populates secrets
+func (s *SecretsPopulator) PopulateSecrets(ctx context.Context, project string, secrets []SecretModel, bucket, keyring, key, location string) error {
+	parentName := fmt.Sprintf("projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s",
+		project, location, keyring, key)
+
+	for _, sec := range secrets {
+		name := fmt.Sprintf("%s.%s", sec.Prefix, fileNameExtension)
+		objHandle := s.storageClient.Bucket(bucket).Object(name)
+		fmt.Printf("Reading object [%s] from bucket [%s]\n", name, bucket)
```

Comment:
Use logger (`logrus`) instead. It will be useful if we will create for example a periodic job or CronJob (like Branch Protector) that will be responsible for syncing secrets.

Reply:
done